lib/matplotlib/tests/test_legend.py

```diff
@@ -1357,3 +1357,10 @@ def test_loc_validation_string_value():
     ax.legend(loc='upper center')
     with pytest.raises(ValueError, match="'wrong' is not a valid value for"):
         ax.legend(loc='wrong')
+
+
+def test_legend_handle_label_mismatch():
+    pl1 = plt.plot(range(10))
+    pl2 = plt.plot(range(10))
+    with pytest.warns(UserWarning, match="number of handles and labels"):
+        plt.legend(handles=[pl1, pl2], labels=["pl1", "pl2", "pl3"])
```

Comment:
We should test that the clipping works:
```suggestion
        legend = plt.legend(handles=[pl1, pl2], labels=["pl1", "pl2", "pl3"])
        assert len(legend.legend_handles) == 2
        assert len(legend.get_texts()) == 2
```